data_structures/hashing/double_hash.py

```diff
@@ -35,6 +35,33 @@ def __hash_double_function(self, key, data, increment):
         return (increment * self.__hash_function_2(key, data)) % self.size_table
 
     def _collision_resolution(self, key, data=None):
+        """
+        Examples:
+
+        1. Try to add three data elements when the size is three
+        >>> dh = DoubleHash(3)
+        >>> dh.insert_data(10)
+        >>> dh.insert_data(20)
+        >>> dh.insert_data(30)
+        >>> dh.keys()
+        {1: 10, 2: 20, 0: 30}
+
+        2. Try to add three data elements when the size is two
+        >>> dh = DoubleHash(2)
+        >>> dh.insert_data(10)
+        >>> dh.insert_data(20)
+        >>> dh.insert_data(30)
+        >>> dh.keys()
+        {10: 10, 9: 20, 8: 30}
+
+        3. Try to add three data elements when the size is one
+        >>> dh = DoubleHash(2)
```

Comment:
```suggestion
        3. Try to add three data elements when the size is one
        >>> dh = DoubleHash(1)
```
Typo

Reply:
> Typo

I tested adding elements to DoubleHash with size 1, to my surprise it resultant in getting stuck in infinite loop, where a new hash value tries to be calculated but is unable to.

For now I have changed that part of test. I'll soon be raising an issue and also will try to fix it.